.gitlab/hpsf-gitlab-ci.yml

```diff
@@ -65,7 +65,8 @@ INTEL-DATA-CENTER-MAX-1100:
     - export ENV_CMAKE_OPTIONS="${ENV_CMAKE_OPTIONS};-D Kokkos_ENABLE_SYCL=ON"
     - export ENV_CMAKE_OPTIONS="${ENV_CMAKE_OPTIONS};-D Kokkos_ARCH_INTEL_PVC=ON"
     - export ENV_CMAKE_OPTIONS="${ENV_CMAKE_OPTIONS};-D Kokkos_ENABLE_TESTS=ON"
-    - export ENV_CMAKE_OPTIONS="${ENV_CMAKE_OPTIONS};-D CMAKE_CXX_FLAGS='-Werror -fsycl-device-code-split=per_kernel -fp-model=precise'"
+    - export ENV_CMAKE_OPTIONS="${ENV_CMAKE_OPTIONS};-D Kokkos_ENABLE_COMPILER_WARNINGS=ON"
+    - export ENV_CMAKE_OPTIONS="${ENV_CMAKE_OPTIONS};-D CMAKE_CXX_FLAGS='-Werror -Wno-zero-length-array -fsycl-device-code-split=per_kernel -fp-model=precise'"
```

Comment:
```
Lock_Array_SYCL-header-b5cd83.h:25:36: warning: zero size arrays are an extension [-Wzero-length-array]
```

Reply:
What that `-header-b5cd83` suffix?

Reply:
So I should not be looking at L25 in `Lock_Array_SYCL.{h,c}pp` files to find out what code that causes that warning?

Reply:
The warning references
```
const char* const kernel_names[] = {
```
that code is not in `Kokkos`.

Reply:
Are push/pop ignore pragmas an option?  How pervasive is that thing?

Reply:
Where would I insert them?

Reply:
I don't where these are emitted from, that's why I am asking you.

Reply:
It seems this file is automatically generated while compiling and I don't have any control over it.

Reply:
Is it only emitted for Desul lock arrays?

Reply:
Yes, we only use global device variables there.

Reply:
Are we able to restrict the suppression to Desul?

Reply:
1eb585c restricts it to the respective source file

Reply:
Actually please confirm that you tried the macro push in the source file and that it didn't not work

Reply:
```diff
diff --git a/tpls/desul/src/Lock_Array_SYCL.cpp b/tpls/desul/src/Lock_Array_SYCL.cpp
index 6660c76e1..9df0ab768 100644
--- a/tpls/desul/src/Lock_Array_SYCL.cpp
+++ b/tpls/desul/src/Lock_Array_SYCL.cpp
@@ -9,6 +9,9 @@ SPDX-License-Identifier: (BSD-3-Clause)
 // FIXME_SYCL Use SYCL_EXT_ONEAPI_DEVICE_GLOBAL when available instead
 #ifdef DESUL_SYCL_DEVICE_GLOBAL_SUPPORTED
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wzero-length-array"
+
 #include <cinttypes>
 #include <desul/atomics/Lock_Array_SYCL.hpp>
 
@@ -59,4 +62,7 @@ void finalize_lock_arrays_sycl<int>(sycl::queue q) {
 }
 
 }  // namespace desul::Impl
+
+#pragma GCC diagnostic pop
+
 #endif
```
doesn't work.